Assets/Scripts/Interaction/ChipInterfaceEditor.cs

```diff
@@ -4,577 +4,608 @@
+    public ChipSignal[][] GetGroups()
+    {
+        var keys = groupsByID.Keys;
+        ChipSignal[][] groups = new ChipSignal[keys.Count][];
+        int i = 0;
+        foreach (var key in keys)
+        {
+            groups[i] = groupsByID[key];
+            i++;
+        }
+        return groups;
     }
-    return true;
-  }
-
-  protected override void FocusLost() {
-    highlightedSignal = null;
-    selectedSignals.Clear();
-    propertiesUI.gameObject.SetActive(false);
-
-    HidePreviews();
-    currentGroupSize = 1;
-  }
-
-  void UpdateUIProperties() {
-    if (selectedSignals.Count > 0) {
-      Vector3 centre =
-          (selectedSignals[0].transform.position +
-           selectedSignals[selectedSignals.Count - 1].transform.position) /
-          2;
-      float propertiesUIX = ScalingManager.propertiesUIX *
-                            (editorType == EditorType.Input ? 1 : -1);
-      propertiesUI.transform.position =
-          new Vector3(centre.x + propertiesUIX, centre.y,
-                      propertiesUI.transform.position.z);
-
-      // Update signal properties
-      for (int i = 0; i < selectedSignals.Count; i++) {
-        selectedSignals[i].UpdateSignalName(nameField.text);
-        selectedSignals[i].useTwosComplement = twosComplementToggle.isOn;
-      }
+
+    // Handles spawning if user clicks, otherwise displays preview
+    void HandleSpawning()
+    {
+
+        if (InputHelper.MouseOverUIObject())
+            return;
+
+        float containerX = chipContainer.position.x +
+                           chipContainer.localScale.x / 2 *
+                               ((editorType == EditorType.Input) ? -1 : 1);
+        float centreY = ClampY(InputHelper.MouseWorldPos.y);
+
+        // Spawn on mouse down
+        if (Input.GetMouseButtonDown(0))
+        {
+            ChipSignal[] spawnedSignals = new ChipSignal[currentGroupSize];
+
+            for (int i = 0; i < currentGroupSize; i++)
+            {
+                float posY = CalcY(InputHelper.MouseWorldPos.y, currentGroupSize, i);
+                Vector3 spawnPos = new Vector3(containerX, posY, chipContainer.position.z + forwardDepth);
+
+                ChipSignal spawnedSignal = Instantiate(signalPrefab, spawnPos, Quaternion.identity, signalHolder);
+                spawnedSignal.GetComponent<IOScaler>().UpdateScale();
+                if (IsGroup)
+                {
+                    spawnedSignal.GroupID = currentGroupID;
+                    spawnedSignal.displayGroupDecimalValue = true;
+                }
+                signals.Add(spawnedSignal);
+                visiblePins.AddRange(spawnedSignal.inputPins);
+                visiblePins.AddRange(spawnedSignal.outputPins);
+                spawnedSignals[i] = spawnedSignal;
+            }
+
+            if (IsGroup)
+            {
+                groupsByID.Add(currentGroupID, spawnedSignals);
+                // Reset group size after spawning
+                currentGroupSize = 1;
```

Comment:
But `currentGroupSize` changes it's value temporarily to handle the spawning of new groups and then reset to 1.

Thus, it is not a reliable source for getting the current selected group size.